docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+
+# Apex-Calcite Integration
+
+Apache Calcite is a highly customizable engine for parsing and planning queries on relational data from various data sources. Apache Calcite provides the functionality of storage independent query optimization and ways to integrate them into other frameworks which would like to take advantage and expose SQL capability to their users. For details about Apache Calcite, please read at [Apache Calcite Website](https://calcite.apache.org/). 
+
+Particularly in SQL on Apex, Calcite processes query and then creates relational algebra to create processing pipelines. These relational algebra processing pipelines are converted to DAG with a set of operators to perform business logic on stream data.
+
+![image alt text](images/calcite-apex.png)
+
+
+Above figure explains how SQL query gets converted to Apex DAG.
+
+1. The received query from a user is parsed by Calcite’s query parser.
+2. The query parser returns the SQL Node Tree to _Query Planner & Optimizer_ which converts the SQL Node Tree to Relation Expression Tree.
+3. This Relation Expression Tree is received by Apache Apex’s SQL module to finally convert to an Apex DAG having series of operators.
+
+One peculiarity that can be seen in above example is that the data source and data destination are not RDBMS systems. This is because Calcite allows Apex to register table sources and destinations as anything which can return a row type results. So a “scan” relational expression gets converted to “KafkaInputOperator + ParseOperator”, a result of which is series of POJOs reflecting a Row Type. Similarly, the “insert” Relational Expression translated to “FormatOperator + FileOutputOperator”.
+
+## SQL APIs for Apache Apex
+
+Listed below are a couple of examples using java APIs which can be used by SQL/Apex users to create a DAG .
+
+## Pure Style SQL Application:
+
+With Apache Calcite Integration, you can also use SQL queries across different data sources and provide UDFs as per your business logic. This example will use Kafka as input source while HDFS file as the output destination.
+Following application code will be used to explain APIs.
+
+     public class PureStyleSQLApplication implements StreamingApplication
+     {
+       @Override
+       public void populateDAG(DAG dag, Configuration conf)
+       {
+         SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+    
+         sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                        "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
```

Comment:
Make "inputTopic" inline with "localhost:9090"... Its a parameter of KafkaEndpoint and not of registerTable.